### PR TITLE
Set version to 0.10.0

### PR DESCRIPTION
**Pull request by gadfort**

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -8,6 +8,34 @@ The changes in each SiliconCompiler release version are described below. Commit
...
 feature are shown in [].
 
+SiliconCompiler 0.10.0 (2023-03-08)
+=========================================
+
+**Major:**
+
+* Schema overhauled (see `schema documentation <https://docs.siliconcompiler.com/en/stable/reference_manual/schema.html>`_ for details):
+
+  * added ability to override certain schema parameters on a per-step/index basis.
```
Comment (aulihant) [inline Changes:18]: Do we want to capitalize the first letter of these sub-bullets?

Comment (gadfort) [reply Changes]: yes, to be consistent

**Comment by aolofsson** [comment]
Please add Aulihan to author list.

**Review by aulihant — APPROVED** [review]
LGTM

**Comment by gadfort** [comment]
@aulihant @aolofsson  great, I'll merge this once the last PR is in for this release

**Pull request merged**